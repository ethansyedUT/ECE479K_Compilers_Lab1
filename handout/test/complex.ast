#17
_program
  #17
  _class
    Main
    Object
    "complex.cl"
    (
    #16
    _method
      main
      Int
      #16
      _block
        #3
        _plus
          #3
          _int
            1
          : Int
          #3
          _int
            2
          : Int
        : Int
        #4
        _sub
          #4
          _int
            4
          : Int
          #4
          _int
            2
          : Int
        : Int
        #5
        _mul
          #5
          _int
            4
          : Int
          #5
          _int
            2
          : Int
        : Int
        #6
        _comp
          #6
          _bool
            0
          : Bool
        : Bool
        #7
        _lt
          #7
          _int
            1
          : Int
          #7
          _int
            2
          : Int
        : Bool
        #8
        _lt
          #8
          _int
            2
          : Int
          #8
          _int
            1
          : Int
        : Bool
        #9
        _leq
          #9
          _int
            1
          : Int
          #9
          _int
            2
          : Int
        : Bool
        #10
        _leq
          #10
          _int
            2
          : Int
          #10
          _int
            2
          : Int
        : Bool
        #11
        _leq
          #11
          _int
            2
          : Int
          #11
          _int
            3
          : Int
        : Bool
        #12
        _eq
          #12
          _int
            2
          : Int
          #12
          _int
            2
          : Int
        : Bool
        #13
        _eq
          #13
          _int
            2
          : Int
          #13
          _int
            1
          : Int
        : Bool
        #14
        _divide
          #14
          _int
            10
          : Int
          #14
          _int
            5
          : Int
        : Int
        #15
        _int
          0
        : Int
      : Int
    )
