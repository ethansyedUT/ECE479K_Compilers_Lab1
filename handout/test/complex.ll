declare i32 @strcmp(i8*, i8*)
declare i32 @printf(i8*, ...)
declare void @abort()
declare i8* @malloc(i32)
define i32 @Main_main() {

main:
	%vtpm.0 = add i32 1, 2
	%vtpm.1 = sub i32 4, 2
	%vtpm.2 = mul i32 4, 2
	%vtpm.3 = icmp slt i32 1, 2
	%vtpm.4 = icmp slt i32 2, 1
	%vtpm.5 = icmp sle i32 1, 2
	%vtpm.6 = icmp sle i32 2, 2
	%vtpm.7 = icmp sle i32 2, 3
	%vtpm.8 = icmp eq i32 2, 2
	%vtpm.9 = icmp eq i32 2, 1
	%vtpm.10 = sdiv i32 10, 5
	ret i32 0
}

@.str = internal constant [25 x i8] c"Main_main() returned %d\0A\00"
define i32 @main() {

entry:
	%vtpm.11 = call i32 @Main_main(  )
	%vtpm.12 = getelementptr [25 x i8], [25 x i8]* @.str, i32 0, i32 0
	%vtpm.13 = call i32(i8*, ... ) @printf( i8* %vtpm.12, i32 %vtpm.11 )
	ret i32 0
}

