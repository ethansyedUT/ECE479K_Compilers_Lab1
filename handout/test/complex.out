Main_main() returned 0
